packages/lit-dev-content/src/components/litdev-search.ts

```diff
@@ -0,0 +1,354 @@
+        color: white;
+      }
+
+      svg .search-icon {
+        fill: white;
+      }
+
+      #padding-container {
+        margin: unset;
+        padding: unset;
+      }
+
+      lion-combobox > lion-options {
+        /* Fix the dimensions of the suggestion dropdown */
+        max-height: min(400px, 100vh - 60px);
+        width: 255px;
+        margin-left: -8px;
+      }
+    }
+  `;
+
+  /**
+   * Text value in search input.
+   */
+  private searchText: string = '';
+
+  /**
+   * Site search index.
+   */
+  private static siteSearchIndex: Minisearch<UserFacingPageData> | null = null;
+
+  /**
+   * Flag so we only load the index once.
+   */
+  private static loadingSearchIndex: boolean = false;
+
+  /**
+   * Search suggestion options.
+   */
+  @queryAll('search-option')
+  private searchOptionElements!: SearchOption[];
+
+  /**
+   * Suggestions visible to the user rendered under the search input field.
+   */
+  @state()
+  private suggestions: Suggestion[] = [];
+
+  /**
+   * Load and deserialize search index into `LitDevSearch.siteSearchIndex`.
+   */
+  async loadSearchIndex () {
+    // We already have a search index.
+    if (LitDevSearch.siteSearchIndex !== null) {
+      return;
+    }
+    if (LitDevSearch.loadingSearchIndex) {
+      return;
+    }
+    LitDevSearch.loadingSearchIndex = true;
+
+    const searchIndexJson = await (await fetch('/searchIndex.json')).text();
+  
+    // Minisearch intialization config must exactly match
+    // `/lit-dev-tools/src/search/plugin.ts` Minisearch options.
+    LitDevSearch.siteSearchIndex = Minisearch.loadJSON<UserFacingPageData>(searchIndexJson, {
+      idField: 'id',
+      fields: ['title', 'heading', 'text'],
+      storeFields: ['title', 'heading', 'relativeUrl', 'isSubsection'],
+      searchOptions: {
+        boost: {title: 1.4, heading: 1.2, text: 1},
+        prefix: true,
+        fuzzy: 0.2,
+      },
+    });
+
+    // If a search query has already been written, fill suggestions.
+    this.querySearch(this.searchText)
+  }
+
+  /**
+   * Load the search index.
+   */
+  firstUpdated() {
+    // Late load search index when page is idle.
+    if ('requestIdleCallback' in window) {
+      (window as any).requestIdleCallback(() => this.loadSearchIndex());
+    } else {
+      setTimeout(() => this.loadSearchIndex(), 500);
+    }
+  }
+
+  /**
+   * Repopulate suggestions with each input event.
+   */
+  handleInput(e: InputEvent) {
+    this.searchText = (e.target as HTMLInputElement).value ?? '';
+    this.querySearch(this.searchText);
+  }
+
+  /**
+   * Populate suggestion dropdown from query.
+   *
+   * An empty query clears suggestions.
+   */
+  querySearch(query: string) {
+    if (!LitDevSearch.siteSearchIndex || query === '' || query.length < 2) {
+      this.suggestions = [];
+      return;
+    }
+    this.suggestions = (LitDevSearch.siteSearchIndex.search(query) ?? []).slice(
+      0,
+      10
+    ) as unknown as Suggestion[];
```

Comment:
This is a great comment. Minisearch appears to return the concrete type `'SearchResult[]'`, leading to the error: `Conversion of type 'SearchResult[]' to type 'Suggestion[]' may be a mistake`. The generic doesn't appear to be used and my guess is that it's because the `search` method takes in a `SearchOptions` that could change the default settings and thus what is returned.